torch/_inductor/codegen/cpp.py

```diff
@@ -1403,28 +1396,20 @@ def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
             f"auto {tmpvar_vec} = at::vec::Vectorized<{DTYPE_TO_CPP[dtype]}>({tmpvar});"
         )
         self.stores.writeline(
-            f"{reduction_combine_vec(reduction_type, tmpvar_vec, value)};"
+            f"{tmpvar_vec} = {reduction_combine_vec(reduction_type, tmpvar_vec, value)};"
         )
 
         if self.tiling_idx >= self.reduction_depth:
             # Horizontal reduction
-            reduce_all_body = "{"
-            if reduction_type == "sum":
-                reduce_all_body += "return x + y;"
-            elif reduction_type == "prod":
-                reduce_all_body += "return x * y;"
-            elif reduction_type == "xor_sum":
-                reduce_all_body += "return x ^ y;"
-            else:
-                reduce_all_body += (
-                    f"return {vec_ns}::{reduce_map[reduction_type]}(x, y);"
-                )
-            reduce_all_body += "}"
+            reduce_all_body = (
+                "{ return " + reduction_combine_vec(reduction_type, "x", "y") + "; }"
+            )
             vec_reduce_all_func = f"{vec_ns}::vec_reduce_all<{DTYPE_TO_CPP[dtype]}>"
-            next_value = f"{vec_reduce_all_func}([]({vec}& x, {vec}&y) {reduce_all_body}, {tmpvar_vec})"
+            next_value = f"{vec_reduce_all_func}([]({vec}& x, {vec}& y) {reduce_all_body}, {tmpvar_vec})"
```

Comment:
Nit. The previous change of passing these by value was potentially beneficial. As these variables fit in registers, they are better passed by value. Perhaps the compiler is able to optimise out this lambda though.